### PR TITLE
:bug: Attach maven identity only when test requires maven settings

### LINKED ISSUE #100: Maven cred is attached even without `requireMavenSettings: true` in the test.yaml file

### PR DESCRIPTION
**Pull request by mguetta1**
Resolves #100 

## Problem

The Tackle Hub target attached a maven settings identity to **every** application whenever maven settings were configured at the target level, regardless of whether the individual test declared `requireMavenSettings: true`. This leaked the maven credential onto tests that never asked for it.

## Fix

Gate both maven-identity attachment sites in `createApplication` on `test.RequireMavenSettings`, matching the kantra target which already checks `test.RequireMavenSettings && k.mavenSettings != ""`.

## Testing

Added `TestTackleHubTarget_CreateApplication_MavenIdentityGating`, which exercises `createApplication` against an in-process `httptest` server and verifies the `/identities` endpoints are only hit when the test requires maven settings. It fails before the fix and passes after.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Maven identity is now attached only when an application explicitly requires Maven settings and those settings are configured.
  - Prevents unnecessary Maven identity requests when Maven settings are available but not required by the test.

### CODE REVIEW

File: pkg/targets/tackle_hub.go [hunk 1]
```diff
@@ -378,8 +378,8 @@ func (t *TackleHubTarget) createApplication(test *config.TestDefinition) (*api.A
 		if existingApp.Name == test.Name {
 			log.Info("Found existing application", "id", existingApp.ID, "name", existingApp.Name)
 
-			// Update identities if maven settings configured
-			if t.mavenSettings != "" {
+			// Update identities only when the test requires maven settings
+			if test.RequireMavenSettings && t.mavenSettings != "" {
```
[the remark this answers is not in the document]
Comment (mguetta1) [reply pkg/targets/tackle_hub.go:382]: Thanks — valid edge case, but its trigger conditions don't occur in this repo today:

- **No duplicate app names.** All 18 `test.yaml` files have unique `name` values (the app name), so within a single `koncur run` no test can inherit another test's persisted maven identity. Each app name maps to exactly one test with one fixed `requireMavenSettings` value.
- **CI runs against a fresh hub.** The `test-hub` flow stands up a fresh Kind cluster + Tackle Hub each run, so there's no carried-over hub state.

The stale-identity path only bites if an existing test's `requireMavenSettings` is flipped `true` → `false` and re-run against a *persistent* hub that already has that app, or if koncur is pointed at a pre-existing hub whose apps already carry a maven identity. Both are real but narrow, and outside the current suite/CI path.

Given that, I'm keeping this PR scoped to gating attachment and will track detach/reconciliation of stale target-managed identities as a possible follow-up rather than expanding this change.

**Review by jmle — APPROVED** [review]

**Comment by mguetta1** [comment]
> Failed to cherry-pick this PR to branch **release-0.9**. [View failed action](https://github.com/konveyor/koncur/actions/runs/34841984333)

https://github.com/konveyor/koncur/pull/121
